components/list/list-item-generic-layout.js

```diff
@@ -56,16 +60,51 @@ class ListItemGenericLayout extends LitElement {
 		`;
 	}
 
+	constructor() {
+		super();
+		this._preventFocus = {
+			handleEvent(event) {
+				event.preventDefault();
+				// target content slot only for now - can add others later
+				// TODO findComposedAncestor should be replaced with event.path || event.composedPath() when supported
+				const slot = findComposedAncestor(event.target, (node) =>
+					node.nodeName === 'SLOT' && ['content'].includes(node.name)
+				);
+				const ancestorSibling = getNextAncestorSibling(slot);
+				const next = getNextFocusable(ancestorSibling, true);
+				// related target is often on the parent
+				const related = getFirstFocusableDescendant(event.relatedTarget);
+				if (!event.relatedTarget) {
+					next.focus();
+				} else {
+					if (event.relatedTarget === next || related === next) {
```

Comment:
Be careful here... given a scenario where focus is being moved from this `button`, say backward tabbing...
```
<my-elem>
    --shadowRoot
        <button>
```
The relatedTarget will be automatically retargeted by the browser to be the element in the context of the handler,  `my-elem`, whereas `next` (in this case) will be the composed DOM element that had the focus, `button`, and this if-statement won't catch it.  It's an edge case, but it could easily come up if people are wrapping focusables in their own components shadowRoots and putting them into the content slot.  If this doesn't get caught it's not a huge deal, but it could make the focus order confusing for screen reader users.

Reply:
Or, could keyboard users get into a focus loop.  Brain 🌫️ ... need more ☕ .

Reply:
Yeah, this was my solution to a focus loop that happened because of what you said above - this _should_ be caught by the calculated `related`, since it refers to the composed item. This happens on the demo page, where `relatedTarget` refers to `d2l-button` but the focusable item is the `button` that lives in the shadow.